bastion/Cargo.toml

```diff
@@ -0,0 +1,27 @@
+[package]
+name = "bastion-amqp"
+version = "0.1.0"
+authors = ["Luca Palmieri <lpalmieri@truelayer.com>"]
+edition = "2018"
+description = "lapin integration with bastion"
+repository = "https://github.com/CleverCloud/lapin"
+readme = "README.md"
+documentation = "https://docs.rs/bastion-amqp"
+keywords = ["amqp", "rabbitmq", "futures", "bastion"]
+categories = ["database"]
+license = "MIT"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[features]
+default = ["lapin/default"]
+
+[dependencies]
+lapin = { version = "^1.0", path = "..", default-features = false }
+bastion = "0.3.4"
+lightproc = "0.3.4"
+bastion-executor = "0.3.4"
```

Comment:
Only bastion seems unused, the two others aren't

Reply:
The `spawn` macro desugars to `bastion_executor::pool::spawn(f, lightproc::proc_stack::ProcStack::default());` which are not re-exported by `bastion`, hence both `bastion_executor` and `lightproc` are needed as dependencies.
We could skip using the `spawn` macro, but I would prefer to avoid relying on its current implementation given that it might change in the future.

Reply:
huh, that's a terrible pattern.
bastion really should reexport those and use the reexports in the macro. That's a *terrible* idea to depend on external crates being available trough macro like this

Reply:
I am inquiring with `bastion`'s maintainers to understand the rationale.

Reply:
Just merged a PR in `bastion` to fix it - once they release `0.3.5` I'll open a PR against `lapin` to remove `bastion-executor` and `lightproc` as direct dependencies.